stetho/src/main/java/com/facebook/stetho/inspector/protocol/module/DOM.java

```diff
@@ -208,6 +214,96 @@ public void run() {
     });
   }
 
+  @ChromeDevtoolsMethod
+  public PerformSearchResponse performSearch(JsonRpcPeer peer, JSONObject params) {
+    final PerformSearchRequest request = mObjectMapper.convertValue(
+        params,
+        PerformSearchRequest.class);
+
+    // Searches are case insensitive so convert
+    // the query to lower case for simpler comparisons.
+    request.query = request.query.toLowerCase();
+
+    final Object root = this.mShadowDOM.getRootElement();
+
+    // Each search action has a unique ID so that
+    // it can be queried later.
+    final String id = UUID.randomUUID().toString();
+
+    final ArrayList<Integer> results = new ArrayList<>();
+    FindMatches(request.query, root, results);
+
+    mSearchResults.put(id, results);
+
+    final PerformSearchResponse response = new PerformSearchResponse();
+    response.searchId = id;
+    response.resultCount = results.size();
+
+    return response;
+  }
+
+  private void FindMatches(String query, Object root, ArrayList<Integer> resultIds) {
```

Comment:
method names should start with lowercase, e.g. `findMatches`. Saw this with `DoesElementMatch`, too.